src/ducks/stripe.duck.js

```diff
@@ -538,7 +538,10 @@ export const createStripeAccount = payoutDetails => (dispatch, getState, sdk) =>
 export const handleCardPayment = params => dispatch => {
   // It's required to use the same instance of Stripe as where the card has been created
   // so that's why Stripe needs to be passed here and we can't create a new instance.
-  const { stripe, card, stripePaymentIntentClientSecret, paymentParams } = params;
+  const { stripe, card, paymentParams, id } = params;
+  const {
+    stripePaymentIntentClientSecret,
+  } = params.attributes.protectedData.stripePaymentIntents.default;
```

Comment:
I think it's better to simplify these params, before this `handleCardPayment` function is called.

```js
const { stripe, card, stripePaymentIntentClientSecret, paymentParams, orderId } = params;
```

This could be done in CheckoutPage.js line 185 (fnParams). `initiateOrder` is currently returning a full transaction entity - so, the comment above that line is wrong:
```js
const fnHandleCardPayment = fnParams => {
      // fnParams should be { stripePaymentIntentClientSecret, stripePaymentIntentId }
```
It should be something like
```js
const fnHandleCardPayment = fnParams => {
      // fnParams should be returned transaction entity
      const order = fnParams;
      const stripePaymentIntentClientSecret = order.attributes.protectedData.stripePaymentIntents.default;
      const orderId = order.id;

      const { stripe, card, billingDetails, paymentIntent } = handlePaymentParams;
      const params = {
        stripePaymentIntentClientSecret,
        orderId,
       // other params that previously were below "...fnParams"
```

Reply:
Btw. we probably should not assume that `protectedData.stripePaymentIntents.default` always exists. Our clients might make strange TX process graphs - and even if they don't make, they might take an update from upstream before their current tx process supports PaymentIntents. 

This dot-chain will throw an exception if `protectedData` or `protectedData.stripePaymentIntents` is missing. Throwing an error might be a correct thing to do, but we should throw it explicitly with a better error message.